backend/src/main/scala/cromwell/backend/validation/TwoKeyRuntimeAttributesValidation.scala

```diff
@@ -0,0 +1,145 @@
+      override def key: String = validation.key
+
+      override def altKey: String = validation.altKey
+
+      override def defaultVal: DefaultValType = validation.defaultVal
+
+      override def coercion: Iterable[WomType] = validation.coercion
+
+      override protected def validateValue: PartialFunction[WomValue, ErrorOr[ValidatedType]] =
+        validation.validateValuePackagePrivate
+
+      override protected def validateExpression: PartialFunction[WomValue, Boolean] =
+        validation.validateExpressionPackagePrivate
+
+      override protected def invalidValueMessage(value: WomValue): String =
+        validation.invalidValueMessagePackagePrivate(value)
+
+      override protected def missingValueMessage: String = s"Expecting $key runtime attribute to be a type in $coercion"
+
+      override def usedInCallCaching: Boolean = validation.usedInCallCachingPackagePrivate
+
+      override protected def staticDefaultOption = Option(default)
+    }
+
+  /**
+   * Returns the value from the attributes matching one of the validation keys. If values are assigned to attribute
+   * matching both keys, the `key` field will override the `altKey` field on the `RuntimeAttributeValidation`.
+   *
+   * Do not use an optional validation as the type internal implementation will throw a `ClassCastException` due to the
+   * way values are located and auto-magically cast to the type of the `runtimeAttributesValidation`.
+   *
+   * @param runtimeAttributesValidation The typed validation to use.
+   * @param validatedRuntimeAttributes  The values to search.
+   * @return The value matching the key.
+   * @throws ClassCastException if the validation is called on an optional validation.
+   */
+  def extractTwoKeys[A, B](runtimeAttributesValidation: TwoKeyRuntimeAttributesValidation[A, B],
+                           validatedRuntimeAttributes: ValidatedRuntimeAttributes
+  ): A = {
+    val key = runtimeAttributesValidation.key
+    val altKey = runtimeAttributesValidation.altKey
+    val primaryValue = extractOption(key, validatedRuntimeAttributes)
+    val altValue = extractOption(altKey, validatedRuntimeAttributes)
+    val value =
+      if (
+        primaryValue.isEmpty || (primaryValue.contains(
+          runtimeAttributesValidation.defaultVal
+        ) && !altValue.isEmpty && !altValue.contains(
+          runtimeAttributesValidation.defaultVal
+        ))
+      ) altValue
+      else primaryValue
+    value match {
+      // NOTE: Some(innerValue) aka Some.unapply() throws a `ClassCastException` to `Nothing$` as it can't tell the type
+      case some: Some[_] => some.get.asInstanceOf[A]
+      case None =>
+        throw new RuntimeException(
+          s"$key not found in runtime attributes ${validatedRuntimeAttributes.attributes.keys}"
+        )
+    }
+  }
+}
+
+trait TwoKeyRuntimeAttributesValidation[A, DefaultValType] extends RuntimeAttributesValidation[A] {
```

Comment:
I'm wondering if you can get away without including `DefaultValType` as a type parameter here. Since `ContinueOnReturnCodeSet` and `ContinueOnReturnCodeFlag` both extend `ContinueOnReturnCode`, can we just use `A` everywhere and not care which `A` implementation the default is?

Reply:
Yes, that is possible!